metrics/storage/fio_job.sh

```diff
@@ -0,0 +1,162 @@
+#  operation (read, write, random-read etc) under certain configuration.
+#  The result is obtained using FIO tool, it runs inside of a container.
+
+set -e
+
+# General env
+SCRIPT_PATH=$(dirname "$(readlink -f "$0")")
+source "${SCRIPT_PATH}/../lib/common.bash"
+
+# Fio job default configuration
+BLOCK_SIZE="4k"
+DEBUG_FILE="debug.log"
+ENGINE="sync"
+FILE_NAME="/home/fio_test"
+FILE_SIZE="4G"
+FIO_IMAGE="clusterhq/fio-tool"
+NUM_JOBS="1"
+OPERATION="randread"
+RUNTIME="cor"
+FIO_TEST_NAME="io_test"
+TEST_NAME="storage fio test"
+TEST_ARGS="$@"
+
+declare FIO_JOB
+
+function help()
+{
+	usage=$(cat << EOF
+Usage: $0 [-h] [options]
+   Description:
+        This script show I/O bandwidth in KB/s
+        of containers using FIO tool.
+   Options:
+        -h,    Help page.
+        -b,    Block size for I/O units.
+        -e,    FIO engine.
+        -f,    Total  size  of  I/O for this job.
+        -o,    I/O operation e.g. read, write etc.
+        -n,    Max number of jobs.
+        -r,    Runtime for docker.
+        -t,    Test name.
+EOF
+)
+	echo "$usage"
+}
+
+# Create the job configuration for FIO
+# this job will run inside of the container.
+function create_fio_job()
+{
+	FIO_JOB="$(cat << EOF
+	#!/bin/bash
+	# This fio job will run inside of
+	# the container.
+
+	fio --ioengine="$ENGINE" \
+		--name="$FIO_TEST_NAME" \
+		--filename="$FILE_NAME" \
+		--bs="$BLOCK_SIZE" \
+		--size="$FILE_SIZE" \
+		--readwrite="$OPERATION" --max-jobs=$NUM_JOBS
+EOF
+)"
+}
+
+# Parse the output of FIO tool in order to get
+# the bandwidth average cosunmed by the FIO job.
+# Also this function will convert results in KB/s
+# if it is necessary.
+function parse_fio_results()
+{
+	raw_data="$1"
+	bw_data=$(echo "$raw_data" |  grep "bw (")
+
+	if [ -z "$bw_data" ];then
+		die "bandwidth result not found: check FIO job configuration"
+	fi
+
+	units=$(echo "$bw_data" grep "bw (" | grep -o "[A-Z]*")
+	avg=$(echo "$bw_data" | awk -F "," '{print $4}' | cut -d "=" -f2)
+	stdev=$(echo "$bw_data" | awk -F "," '{print $5}' | cut -d "=" -f2)
+
+	echo "Units: $units"
+	case "$units" in
+		MB)
+		    avg=$(echo "$avg * 1024" | bc -l)
+		    ;;
+		GB)
+		    avg=$(echo "($avg * 1024) * 1024" | bc -l)
+		    ;;
+		esac
+
+	save_results "$TEST_NAME" "$TEST_ARGS" "$avg" "KB/s"
+	echo "$avg KB/s"
+}
+
+function main()
+{
+	cmds=("bc" "awk")
+	local OPTIND
+	while getopts "b:e:hs:o:n:r:t:" opt;do
+		case ${opt} in
+		b)
+		    BLOCK_SIZE="${OPTARG}"
+		    ;;
+		e)
+		    ENGINE="${OPTARG}"
+		    ;;
+		h)
+		    help
+		    exit 0;
+		    ;;
+		s)
+		    FILE_SIZE="${OPTARG}"
+		    ;;
+		o)
+		    OPERATION="${OPTARG}"
+		    ;;
+		n)
+		    NUM_JOBS="${OPTARG}"
+		    ;;
+		r)
+		    RUNTIME="${OPTARG}"
+		    ;;
+		t)
+		    TEST_NAME="${OPTARG}"
+		    ;;
+		esac
+	done
+	shift $((OPTIND-1))
+
+	check_cmds "${cmds[@]}"
+	init_env
+	create_fio_job
+
+	# Launch container (cor runtime by default)
+	output=$(docker run --runtime="$RUNTIME" \
```

Comment:
Just a note - I'm OK with us here using the default path/system docker, as that is generally where we have installed the runtime we are testing. But, just to note, possibly in the future we may have to make this a `$DOCKER` or similar that will be set by the metrics framework.